metsrw/premis_object.py

```diff
@@ -0,0 +1,146 @@
+        self.filename = filename
+        self.characteristics = characteristics
+        self.identifier_type = identifier_type
+        self.identifier_value = identifier_value
+        self.relationship_type = relationship_type
+        self.relationship_subtype = relationship_subtype
+
+    @classmethod
+    def parse(cls, root):
+        """
+        Create a new Object by parsing root.
+
+        :param root: Element or ElementTree to be parsed into an object.
+        :raises exceptions.ParseError: If the root is not object.
+        """
+        if root.tag != utils.lxmlns('premis') + 'object':
+            raise exceptions.ParseError('Object can only parse object elements with PREMIS namespace.')
+
+        filename = os.path.basename(root.find(utils.lxmlns('premis') + "originalName").text)
+
+        # Parse characteristics
+        characteristics_el = root.find(utils.lxmlns('premis') + "objectCharacteristics")
+        characteristics = ObjectCharacteristics.parse(characteristics_el)
+
+        # Parse identifier
+        identifier_el = root.find(utils.lxmlns('premis') + "objectIdentifier")
+
+        if identifier_el is not None:
+            identifier_type_el = identifier_el.find(utils.lxmlns('premis') + "objectIdentifierType")
+            identifier_type = identifier_type_el.text if identifier_type_el is not None else None
+
+            identifier_value_el = identifier_el.find(utils.lxmlns('premis') + "objectIdentifierValue")
+            identifier_value = identifier_value_el.text if identifier_value_el is not None else None
+        else:
+            identifier_type = None
+            identifier_value = None
+
+        # Parse relationship to other objects
+        relationship_el = root.find(utils.lxmlns('premis') + "relationship")
+
+        if relationship_el is not None:
+            type_el = relationship_el.find(utils.lxmlns('premis') + "relationshipType")
+            relationship_type = type_el.text if type_el is not None else None
+
+            subtype_el = relationship_el.find(utils.lxmlns('premis') + "relationshipSubType")
+            relationship_subtype = subtype_el.text if subtype_el is not None else None
+        else:
+            relationship_type = None
+            relationship_subtype = None
+
+        return cls(filename, characteristics, identifier_type, identifier_value, relationship_type, relationship_subtype)
+
+
+class ObjectCharacteristics(object):
+    """
+    An object representing a PREMIS objectCharacteristics element.
+
+    :param str size: File size in bytes.
+    :param str format_name: Name of the file format. TODO
+    :param str format_version: Version of the file form.
+    :param str format_registry_name: Name of the format registry.
+    :param str format_registry_key: Registry-specific format key.
+    :param boolean is_mets: Whether or not the file is a METS file.
```

Comment:
Putting the object characteristics in another class is just a way to break up the functionality into less huge pieces.